bot/exts/advent_of_code/_cog.py

```diff
@@ -385,28 +385,49 @@ async def aoc_leaderboard(self, ctx: commands.Context, *, aoc_name: str | None =
             aoc_name = aoc_name[1:-1]
 
         # Check if an advent of code account is linked in the Redis Cache if aoc_name is not given
+        is_explict_name: bool = bool(aoc_name)
         if (aoc_cache_name := await _caches.account_links.get(ctx.author.id)) and aoc_name is None:
             aoc_name = aoc_cache_name
 
+        name_not_found: bool = False
```

Comment:
same here

Reply:
@onerandomusername That still doesn't explain why the annotation would be needed. What would we lose if it was removed?

Besides, pyright ignores explicit annotations on local variables
```py
def foo():
    x: bool = False
    reveal_type(x)  # still Literal[False]
```